src/iteration.jl

```diff
@@ -2,76 +2,85 @@
 Tables.rowaccess(::Type{<:File}) = true
 Tables.rows(f::File) = f
 
-struct Row{T}
-    file::File
-    row::T
+@inline Base.@propagate_inbounds function Base.getindex(f::File{false}, row::Int)
+    @boundscheck checkbounds(f, row)
+    return Row{false}(getnames(f), getcolumns(f), getlookup(f), row, 0, 0)
 end
 
-getfile(r::Row) = getfield(r, :file)
-getrow(r::Row) = getfield(r, :row)
-
-Base.propertynames(r::Row) = getnames(getfile(r))
+@inline Base.@propagate_inbounds function Base.getindex(f::File{true}, row::Int)
+    @boundscheck checkbounds(f, row)
+    c = getcolumn(f, 1)
+    i = row
+    for (j, A) in enumerate(c.args)
+        n = length(A)
+        i <= n && return Row{true}(getnames(f), getcolumns(f), getlookup(f), row, j, i)
+        i -= n
+    end
+    return Row{true}(getnames(f), getcolumns(f), getlookup(f), row, 1, 1)
+end
 
-function Base.show(io::IO, r::Row)
-    println(io, "CSV.Row($(getrow(r))) of:")
-    show(io, getfile(r))
+# non-threaded file
+@inline function Base.iterate(f::File{false}, st::Int=1)
```

Comment:
This implementation should be free from `AbstractArray`.

Reply:
Ah, you're probably right; but I do want to provide an overload for the `threaded` case, because we can be more efficient.

Reply:
That's fine; there won't be a method ambiguity.